### DISCUSSION TITLE
how do i use font awesome with a font import

### DISCUSSION THREAD

**Discussion by Tricked-dev** [slint-ui/slint#3263] [Q&A] [answered]
i currently have this:

```js
import "../assets/fa-regular-400.ttf";
import "../assets/Font Awesome 6 Brands-Regular-400.otf";
import "../assets/Font Awesome 6 Free-Regular-400.otf";
```

and then

```slint

export component AppWindow inherits Window { 
  default-font-family: "Font Awesome";
  Text {
     font-family: "Font Awesome";
     text: @tr(" test");
  }
}
```
this doesn't work sadly.

when i install the font as a system font it works but that's obviously not what i want when shipping the application

how could i fix this / what other methods are there to get icons?

**Comment by tronical** [comment, marked as answer]
Hm, is it possible that you're using the wrong family name?

I tried this and it works for me:

1. Create a file `test.slint` with the following contents:

```
import "fa-regular-400.ttf";

export component AppWindow inherits Window {
  default-font-size: 48px;
  Text {
     font-family: "Font Awesome 6 Free";
     text: @tr("test");
  }
}
```

2. Download `fa-regular-400.ttf` from https://github.com/FortAwesome/font-awesome-sass/tree/master/assets/fonts/font-awesome
3. Preview with `slint-viewer test.slint` 

I see this on macOS. I don't have font awesome installed on my system.

<img width="186" alt="Screenshot 2023-08-11 at 13 36 53" src="https://github.com/slint-ui/slint/assets/1486/1bcf3108-9f33-4c5f-8c8d-971f54f3c387">

**Reply by Tricked-dev** [reply]
ah thats it thanks! although something like this `default-font-family: "Roboto, Font Awesome 6 Free";` sadly isnt possible so either everything is in the weird fa font or i have to do some annoying text stuff

**Reply by Tricked-dev** [reply]
update: i used nerdfonts to generate a font based on lato with all the unicode emojis :tada:
